weave

```diff
@@ -238,14 +239,17 @@ elif [ "$1" = "run" -a -z "$IS_LOCAL" ] ; then
     # because we are passing arbitrary arguments on to docker run, and
     # we can't rely on our baked-in docker to support those arguments.
     shift 1
+    ATTACH_ARGS="--or-die"
     if [ "$1" != "--without-dns" ] ; then
         DNS_ARGS=$(exec_remote dns-args "$@" || true)
+        DNS_EXTRA_HOSTS_ARGS=$(exec_remote dns-extra-hosts "$@" || true)
+        ATTACH_ARGS="$ATTACH_ARGS --rewrite-hosts"
     fi
     shift $(dns_arg_count "$@")
     collect_cidr_args "$@"
     shift $CIDR_ARG_COUNT
-    CONTAINER=$(docker $DOCKER_CLIENT_ARGS run -e WEAVE_CIDR=none $DNS_ARGS -d "$@")
-    exec_remote attach $CIDR_ARGS --or-die $CONTAINER >/dev/null
+    CONTAINER=$(docker $DOCKER_CLIENT_ARGS run -e WEAVE_CIDR=none $DNS_ARGS $DNS_EXTRA_HOSTS_ARGS -d "$@")
```

Comment:
why do we have `$DNS_EXTRA_HOSTS_ARGS` in here? They will be in `$@` too, surely.